setup/command/apply.go

```diff
@@ -16,15 +16,24 @@ func (f *CommandFactory) Apply() cli.Command {
 				Name:  "quick",
 				Usage: "skips verification checks that normally run after 'terraform apply' has completed",
 			},
+			cli.BoolTFlag{
+				Name:  "push",
+				Usage: "setting it to false skips pushing local tfstate to s3 (default: true)",
+			},
 		},
 		Action: func(c *cli.Context) error {
 			args, err := extractArgs(c.Args(), "NAME")
 			if err != nil {
 				return err
 			}
 
+			provider, err := f.newAWSProviderHelper(c)
```

Comment:
Ok, so in order for `push` or `pull`  commands to work, the user needs to give credentials for AWS. This can be done through cli flags (the `awsFlags` mentioned earlier), or environment variables.  What we used to do in the old l0-setup was grab the aws keys from whatever they entered during [the equivalent of] `l0-setup init`. 

This is nice for the user, but introduced a ton of complexities and bad coupling patterns in the codebase.
I intentionally took out that behavior so we don't have to deal with it anymore. 

But now we have a UX issue - if we `push` after every `apply`, then user needs to make sure they input aws credentials every time they run `apply`. This seems acceptable for `push`, but I could see customers complaining about `apply`.